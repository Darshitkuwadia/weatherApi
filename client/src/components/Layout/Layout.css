.layout {
  height: 100vh;
  width: 100vw;
  background-color: var(--color-white);
}

.header {
  width: 100%;
  text-align: center;
  padding: 25px 0;
  color: var(--color-white);
  background-color: var(--color-brand);
  border-bottom: 1px solid var(--color-grey2);
}

.header h1{
  margin: 0;
}

.content {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}