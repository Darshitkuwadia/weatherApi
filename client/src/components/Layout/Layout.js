import React from 'react';
import './Layout.css';

function Layout(props) {
  const {
    title,
    subTitle,
    children,
  } = props

  return (
    <div className="layout">
      <div className="header">
        <h1>{title}</h1>
        <h3>{subTitle}</h3>
      </div>
      <div className="content">
        {children}
      </div>
    </div>
  )
}

export default Layout