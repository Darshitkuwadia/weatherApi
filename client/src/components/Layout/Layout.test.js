import React from 'react';
import { shallow } from 'enzyme';
import Layout from './Layout';

const title = 'Layout Title';
const subTitle = 'Layout Sub Title';

describe('Layout', () => {
  it('should render the Title amd SubTitle Component correctly', () => { 
    let layout = shallow(
      <Layout title={title} subTitle={subTitle}/> 
    )  
    expect(layout).toMatchSnapshot();
  });
  it('renders the children', () => {
    let layout = shallow(
      <Layout title={title} subTitle={subTitle}>
        <p>Children</p>
      </Layout>
    ); 
    expect(layout).toMatchSnapshot();
  });
}); 