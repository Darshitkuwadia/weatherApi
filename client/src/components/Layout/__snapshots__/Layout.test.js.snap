// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders the children 1`] = `
<div
  className="layout"
>
  <div
    className="header"
  >
    <h1>
      Layout Title
    </h1>
    <h3>
      Layout Sub Title
    </h3>
  </div>
  <div
    className="content"
  >
    <p>
      Children
    </p>
  </div>
</div>
`;

exports[`Layout should render the Title amd SubTitle Component correctly 1`] = `
<div
  className="layout"
>
  <div
    className="header"
  >
    <h1>
      Layout Title
    </h1>
    <h3>
      Layout Sub Title
    </h3>
  </div>
  <div
    className="content"
  />
</div>
`;
